http/src/main/resources/jupyter/init-actions.sh

```diff
@@ -452,10 +452,15 @@ END
 
       log 'Starting Jupyter Notebook...'
       retry 3 docker exec -d ${JUPYTER_SERVER_NAME} ${JUPYTER_SCRIPTS}/run-jupyter.sh ${NOTEBOOKS_DIR}
-      log 'All done!'
 
       STEP_TIMINGS+=($(date +%s))
     fi
+
+    # Remove any unneeded cached images to save disk space.
+    # Do this asynchronously so it doesn't hold up cluster creation
+    log 'Pruning docker images...'
+    docker image prune -a -f &
```

Comment:
I did this manually -- it took about a minute and reclaimed 18.49GB of disk on the machine.